hazelcast/src/main/java/com/hazelcast/spi/NamedOperation.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi;
+
+public interface NamedOperation {
```

Comment:
Perhaps it is better to let the service decides: you ask the service if he can check whatever needs to be checked. Now you need to expose some internal detail of a service (e.g. having a map) to implement the quorum.